src/NuGet.Commands/RestoreCommand.cs

```diff
@@ -538,6 +544,29 @@ private bool ResolutionSucceeded(IEnumerable<RestoreTargetGraph> graphs)
                         correctedPackageName: library.Name);
 
                     target.Libraries.Add(targetLibrary);
+
+                    // Log warnings if the target library used the fallback framework
+                    if (warnForImportsOnGraph && !librariesWithWarnings.Contains(library))
```

Comment:
Wouldn't this warn if the target framework says don't warn i.e. one tfm says warn and the other doesn't?